cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/connector/resource/AzureStorageAccountBuilderService.java

```diff
@@ -0,0 +1,42 @@
+package com.sequenceiq.cloudbreak.cloud.azure.connector.resource;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.microsoft.azure.management.resources.Deployment;
+import com.sequenceiq.cloudbreak.cloud.azure.AzureStorageAccountTemplateBuilder;
+import com.sequenceiq.cloudbreak.cloud.azure.client.AzureClient;
+import com.sequenceiq.cloudbreak.cloud.exception.CloudConnectorException;
+import com.sequenceiq.cloudbreak.common.json.Json;
+
+@Service
+public class AzureStorageAccountBuilderService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureStorageAccountBuilderService.class);
+
+    @Inject
+    private AzureStorageAccountTemplateBuilder azureStorageAccountTemplateBuilder;
+
+    public void buildStorageAccount(AzureClient client, StorageAccountParameters storageAccountParameters) {
+
+        try {
+            String template = azureStorageAccountTemplateBuilder.build(storageAccountParameters);
+            if (!client.templateDeploymentExists(storageAccountParameters.getResourceGroupName(), storageAccountParameters.getStorageAccountName())) {
+                String parameters = new Json(Map.of()).getValue();
+                Deployment templateDeployment = client.createTemplateDeployment(
+                        storageAccountParameters.getResourceGroupName(), storageAccountParameters.getStorageAccountName(), template, parameters);
+                LOGGER.debug("Created template deployment for storage account: {}", templateDeployment.exportTemplate().template());
+            }
+        } catch (Exception e) {
```

Comment:
We should use the same mechanism as the other occurrences, e.g: https://github.com/hortonworks/cloudbreak/blob/master/cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/upscale/AzureUpscaleService.java#L92-L102